### PR TITLE
Reland: Reword flutter run help screen.

### PR DESCRIPTION
**Pull request by zanderso**
This is an attempt to reland https://github.com/flutter/flutter/pull/44106

### CODE REVIEW

File: packages/flutter_tools/lib/src/run_hot.dart [hunk 1]
```diff
@@ -1045,29 +1045,26 @@ class HotRunner extends ResidentRunner {
...
       printHelpDetails();
-      globals.printStatus('To repeat this help message, press "h". $quitMessage');
-    } else {
-      globals.printStatus('For a more detailed help message, press "h". $quitMessage');
+    }
+    for (final FlutterDevice device in flutterDevices) {
+      final String dname = device.device.name;
+      // Caution: This log line is parsed by device lab tests.
+      globals.printStatus(
+        'An Observatory debugger and profiler on $dname is available at: '
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/run_hot.dart]: The original PR introduced a newline at the end of this line, which confused some parsing in a few device lab tests. I've removed the newline.

**Comment by zanderso** [comment]
@esarbanis if landing your commits in this PR is alright with you, could you please respond with "@googlebot I consent."? Thanks!

**Comment by esarbanis** [comment]
@googlebot I consent.

**Review by jonahwilliams — APPROVED** [review]
LGTM!

Have you run the same test locally to verify?

**Comment by zanderso** [comment]
Found a couple more cases with some further grepping. The tests that were failing now all pass for me locally.

**Comment by esarbanis** [comment]
@zanderso thanks for driving this to completion!

**Pull request merged**